src/silo/common/range.cpp

```diff
@@ -0,0 +1,77 @@
+#include "silo/common/range.h"
+
+#include <cstddef>
+
+#include <fmt/format.h>
+#include <boost/numeric/conversion/cast.hpp>
+
+namespace silo::common {
+
+template <typename T>
+Range<T>::Range(T first, T beyond_last)
+    : first_(first),
+      beyond_last_(beyond_last) {
+   if (!(first <= beyond_last)) {
```

Comment:
```suggestion
   if (first > beyond_last) {
```

Reply:
My reason for stating it that way is that "first <= beyond_last" is the precondition that needs to be upheld, and I try not to convert anything--especially given the dangers of number bounds I try to be as stupid as possible.